src/v1/album.js

```diff
@@ -15,33 +17,49 @@ router.get('/', (req, res) => {
   });
 });
 
+router.get('/:id(\\w{24})', (req, res) => {
+  Album.getAlbum(req.params.id)
+    .then(album => {
+      res.status(200).send(new Response.Data(album));
+    });
+});
+
 router.post('/', (req, res) => {
   if (!req.body.name || req.body.name == '') {
     res.status(400)
       .send(new Response.Error(new Err.InvalidParam(['name is required'])));
-  }
-
-  if (req.body.artist && req.body.artist != '') {
-    Artist.getArtist(req.body.artist)
-      .then(artist => {
-        new Album(req.body.name, artist.toSimple()).save()
-          .then(album => {
-            artist.albums.push(album.toSimple());
-            artist.save().then(artist => res.status(200).send(new Response.Data(artist)));
-          });
-      });
   } else {
     new Album(req.body.name).save().then(album => {
       res.status(200).send(new Response.Data(album));
     });
   }
 });
 
-router.get('/:id(\\w{24})', (req, res) => {
-  Album.getAlbum(req.params.id)
-    .then(album => {
-      res.status(200).send(new Response.Data(album));
-    });
+router.delete('/:id(\\w{24})', (req, res) => {
+  /** check constraint */
+  Album.getAlbum(req.params.id).then(album => {
+    if (!album) {
+      res.status(400)
+        .send(new Response.Error(new Err.InvalidParam(['album not found'])));
+    } else if (album.tracks && album.tracks.length > 0) {
+      res.status(400)
+        .send(new Response.Error((new Err.UnremovableError('has related tracks'))));
+    } else if (album.artist) {
```

Comment:
I think this is the interesting problem here. We don't delete the album which has tracks, because those tracks will be orphan once the album is deleted.
But I don't think the artist has the same problem here, right?

Reply:
I remove the tracks related check. =)